metrics/src/main/java/com/expedia/adaptivealerting/metrics/functions/service/MetricQueryService.java

```diff
@@ -0,0 +1,98 @@
+package com.expedia.adaptivealerting.metrics.functions.service;
+
+import com.expedia.adaptivealerting.anomdetect.util.HttpClientWrapper;
+import com.expedia.adaptivealerting.metrics.functions.source.MetricFunctionsSpec;
+import com.expedia.adaptivealerting.metrics.functions.source.graphite.GraphiteQueryResult;
+import com.expedia.adaptivealerting.metrics.functions.source.graphite.ConstructSourceURI;
+import com.expedia.metrics.MetricData;
+import com.expedia.metrics.MetricDefinition;
+import com.expedia.metrics.TagCollection;
+import com.typesafe.config.Config;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.client.fluent.Content;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+@Slf4j
+public class MetricQueryService {
+    private HttpClientWrapper metricFunctionHttpClient;
+    private final String EMPTY_RESULT_FROM_SOURCE = "{}";
+    private final String METRICDATA_KEY_FOR_EMPTRY_DATA = "aggregator.producer.";
+    private final double METRICDATA_DEFAULT_VALUE = 0.0;
+    private final String METRIC_SOURCE_KEY = "metric-source";
+    /* For now source supported is graphite alone , non-graphite is just
+    *  a place holder for the common method
+    */
+    private final List<String> METRIC_SOURCE = Arrays.asList("graphite", "non-graphite");
+    private final String GRAPHITE_SOURCE_KEY = "graphite";
+    private final static String GRAPHITE_KEY_TAG = "name";
+
+    public MetricQueryService(HttpClientWrapper httpClientWrapper) {
+        metricFunctionHttpClient = httpClientWrapper;
+    }
+
+    /* common method for generic metric source */
+    public MetricData getMetricQueryResult(Config metricSourceSinkConfig, MetricFunctionsSpec metricFunctionsSpec) {
+        if (METRIC_SOURCE.contains(metricSourceSinkConfig.getString(METRIC_SOURCE_KEY))) {
+            return queryMetricSource(metricSourceSinkConfig, metricFunctionsSpec);
+        }
+        else {
+            return defaultMetricData(metricFunctionsSpec);
+        }
+    }
```

Comment:
This if/else is an example of the sort of thing I wouldn't want to do, because it violates the OCP as I noted above. Please assume that this is Graphite-only right now, and that whatever Graphite-only code you're implementing will become an SPI plugin if we ever generalize, meaning that we still won't need support for non-Graphite inside this code.

Reply:
In fact I'd suggest renaming this class to reflect the fact that it specifically targets Graphite: say `GraphiteMetricFunctions` or similar.